src/Externals/Interfaces/ORM/EntityManagerInterface.php

```diff
@@ -0,0 +1,45 @@
+<?php
+declare(strict_types=1);
+
+namespace LoyaltyCorp\Multitenancy\Externals\Interfaces\ORM;
+
+use EoneoPay\Externals\ORM\Interfaces\EntityInterface;
+
+interface EntityManagerInterface
+{
+    /**
+     * Flush unit of work to the database, ensuring all entities belong to the correct provider if applicable
+     *
+     * @param int $providerId The provider id to match against unit of work
+     *
+     * @return void
+     */
+    public function flush(int $providerId): void;
```

Comment:
As discussed: Maybe this should be a Provider argument instead.

Reply:
Having missed the discussion, I think supporting both might be a good idea - avoiding hydration of the provider when we can makes a lot of sense..

Reply:
@merk it will only use the proxy because we're using the primary id